src/ol/reproj/Triangulation.js

```diff
@@ -138,11 +139,16 @@ class Triangulation {
     const sourceBottomRight = this.transformInv_(destinationBottomRight);
     const sourceBottomLeft = this.transformInv_(destinationBottomLeft);
 
+    const maxSubdivision = MAX_SUBDIVISION + (opt_destinationResolution ?
+      Math.max(0, Math.ceil(Math.log2(getArea(targetExtent) /
+        (opt_destinationResolution * opt_destinationResolution * 256 * 256))))
```

Comment:
Why 256?

Reply:
I guess maybe this calculation assumes the `MAX_SUBDIVISION` value was estimated for the 256x256 tiles, so this calculates how much "larger" this extent is than that?

Anyway, it would be great to describe this calculation in more detail (or maybe separate into more variables with more descriptive names?)..